File: Structural/Lab10/Typescript/src/Entities/Decorators/augmented_reality_decorator.ts
import { IRealEstateExperience } from "../../Interfaces/real_estate_experience";
import { RealEstateExperienceDecorator } from "./real_estate_experience_decorator";

export class AugmentedRealityDecorator extends RealEstateExperienceDecorator {

    constructor(realEstate: IRealEstateExperience) {
        super(realEstate);
    }
    public customizeExperience(): String {
        return this._realEstate.customizeExperience() + "\nFeatures augmented reality to visualize furniture and decor.";
    }
}

File: Structural/Lab13/C#/Entities/Decorators/AugmentedRealityDecorator.cs
using System;
using System.Collections.Generic;
using System.Text;
using C_.Interfaces;

namespace C_.Entities.Decorators
{
    public class AugmentedRealityDecorator : RealEstateExperienceDecorator
    {
        public AugmentedRealityDecorator(IRealEstateExperience realEstate) : base(realEstate) { }

        public override string CustomizeExperience()
        {
            return _realEstate.CustomizeExperience() + "\nFeatures augmented reality to visualize furniture and decor.";
        }
    }
}
